### PR TITLE
Extract `extract_schema_qualified_name` method

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 1]
```diff
@@ -523,6 +521,10 @@ def add_index(table_name, column_name, options = {}) #:nodoc:
       end
 
       def foreign_keys(table_name)
+        raise ArgumentError unless table_name.present?
```
Comment (arthurnn) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: Is this condition possible?
I guess we need a regression test for this now.

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 2]
```diff
@@ -530,8 +532,8 @@ def foreign_keys(table_name)
                 ,fk.constraint_name as 'name'
           FROM information_schema.key_column_usage fk
           WHERE fk.referenced_column_name is not null
-            AND fk.table_schema = '#{@config[:database]}'
-            AND fk.table_name = '#{table_name}'
+            AND fk.table_schema = #{quote(schema)}
+            AND fk.table_name = #{quote(name)}
```
Comment (arthurnn) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: why this change? is this fixing a bug, changing any current behaviour?

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb (hunk 1 above)
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: @jeremy said he prefer to raise an ArgumentError for an exceptional case like this in https://github.com/rails/rails/pull/21614#discussion_r39651518

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb (hunk 2 above)
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: For consistency.

data_source_exists?:

```ruby
        sql = "SELECT table_name FROM information_schema.tables "
        sql << "WHERE table_schema = #{quote(schema)} AND table_name = #{quote(name)}"
```

view_exists?:

```ruby
        sql = "SELECT table_name FROM information_schema.tables WHERE table_type = 'VIEW'"
        sql << " AND table_schema = #{quote(schema)} AND table_name = #{quote(name)}"
```

primary_keys:

```ruby
        select_values(<<-SQL.strip_heredoc, 'SCHEMA')
          SELECT column_name
          FROM information_schema.key_column_usage
          WHERE constraint_name = 'PRIMARY'
            AND table_schema = #{quote(schema)}
            AND table_name = #{quote(name)}
          ORDER BY ordinal_position
        SQL
```

**Comment by arthurnn** [comment]
at first look code looks fine.. thoughts @jeremy ?

**Comment by kamipo** [comment]
@jeremy What do you think about this?

**Comment by sgrif** [comment]
Please don't repeatedly ping maintainers, once it's been assigned it will be reviewed when they have time.
